thrift/admin.thrift

```diff
@@ -251,6 +251,20 @@ service AdminService {
     throws (
       1: shared.InternalServiceError internalServiceError,
     )
+
+  void DeleteWorkflow(1: AdminDeleteWorkflowRequest request)
+    throws (
+      1: shared.BadRequestError         badRequestError,
+      2: shared.EntityNotExistsError    entityNotExistError,
+      3: shared.InternalServiceError    internalServiceError,
```

Comment:
nit: please remove 5XX type errors from expected error defintions as there's no case where this is a valid state. In practice I realised this an admin API and it shouldn't matter, but it makes tracking down errors much more difficult.

Reply:
This logic is shared across our thrift files. If I remember this correctly this something to do with our Java client where it needs to know which type of errors are being thrown and InternalServiceError is the wrapper for all the undefined errors.